examples/ASDKgram/Sample/WindowWithStatusBarUnderlay.m

```diff
@@ -0,0 +1,40 @@
+//
+//  WindowWithStatusBarUnderlay.m
+//  Sample
+//
+//  Created by Hannah Troisi on 4/10/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import "WindowWithStatusBarUnderlay.h"
+#import "Utilities.h"
+
+@implementation WindowWithStatusBarUnderlay
+{
+  UIView *_statusBarOpaqueUnderlayView;
+}
+
+-(instancetype)initWithFrame:(CGRect)frame
+{
+  self = [super initWithFrame:frame];
+  if (self) {
+    _statusBarOpaqueUnderlayView                 = [[UIView alloc] init];
+    _statusBarOpaqueUnderlayView.backgroundColor = [UIColor darkBlueColor];
+    [self addSubview:_statusBarOpaqueUnderlayView];
+  }
+  return self;
+}
+
+-(void)layoutSubviews
+{
+  [super layoutSubviews];
+  
+  [self bringSubviewToFront:_statusBarOpaqueUnderlayView];
+  
+  _statusBarOpaqueUnderlayView.frame = CGRectMake(0,
+                                                  0,
+                                                  [[UIScreen mainScreen] bounds].size.width,
```

Comment:
Probably worth a comment why this is necessary rather than using the status bar frame - I assume it is not correct during the moment of rotation or something?  Is this one always the right one?  Do you need to switch between width and height depending on orientation?

Might be simpler to read to start by setting rect = statusbarframe, then change rect.size.width =